src/Integration.Vsix.UnitTests/SonarLintTagger/TextBufferIssueTrackerTests.cs

```diff
@@ -253,231 +261,110 @@ private void CheckAnalysisWasRequested()
+        public void WhenFileIsRenamed_FileHasNoAssociatedProject_NoExceptionIsThrown()
         {
-            mockSonarErrorDataSource.Invocations.Clear();
-
-            // Arrange
-            var filteredIssue = CreateIssue("single issue", startLine: 1, endLine: 1);
+            mockSolution.Reset();
+            mockSolution
+                .Setup(x => x.FindProjectItem(mockedJavascriptDocumentFooJs.Name))
+                .Returns((ProjectItem)null);
 
-            var issuesToReturnFromFilter = Enumerable.Empty<IAnalysisIssueVisualization>();
-            SetupIssuesFilter(out var capturedFilterInput, issuesToReturnFromFilter);
+            Action act = () => mockedJavascriptDocumentFooJs.Raise(x => x.FileActionOccurred += null, new TextDocumentFileActionEventArgs(
+                mockedJavascriptDocumentFooJs.Name, DateTime.Now,
+                FileActionTypes.DocumentRenamed));
 
-            // Act
-            testSubject.HandleNewIssues(new [] { filteredIssue });
+            act.Should().NotThrow();
 
-            // Assert
-            // Check the expected issues were passed to the filter
-            capturedFilterInput.Count.Should().Be(1);
-            capturedFilterInput[0].Should().Be(filteredIssue);
+            logger.AssertNoOutputMessages();
+        }
 
-            CheckErrorListRefreshWasRequestedOnce(testSubject.Factory);
+        #endregion
 
-            // Check there are no issues
-            testSubject.Factory.CurrentSnapshot.Issues.Count().Should().Be(0);
-        }
+        #region RequestAnalysis
 
         [TestMethod]
-        public void WhenNewFileLevelIssuesAreFound_IssueNotFiltered_ListenersAreUpdated()
+        public void Create_AnalysisIsRequestedOnCreation()
         {
-            mockSonarErrorDataSource.Invocations.Clear();
-
-            // Arrange
-            var issues = new[] { CreateFileLevelIssue("File Level Issue") };
+            var issueConsumer = new Mock<IIssueConsumer>();
+            var issueConsumerFactory = CreateIssueConsumerFactory(issueConsumer.Object);
 
-            SetupIssuesFilter(out var capturedFilterInput, issues);
+            testSubject = CreateTextBufferIssueTracker(issueConsumerFactory: issueConsumerFactory.Object);
 
-            // Act
-            testSubject.HandleNewIssues(issues);
+            // The factory should have been called...
+            issueConsumerFactory.Verify(x => x.Create(mockedJavascriptDocumentFooJs.Object, It.IsAny<string>(), It.IsAny<Guid>(), It.IsAny<PublishSnapshot>()), Times.Once);
 
-            // Assert
-            // Check the expected issues were passed to the filter
-            capturedFilterInput.Count.Should().Be(1);
-            capturedFilterInput.Should().BeEquivalentTo(issues);
-
-            CheckErrorListRefreshWasRequestedOnce(testSubject.Factory);
-            
-            testSubject.Factory.CurrentSnapshot.Issues.Count().Should().Be(1);
+            // and the return issue consumer should have been cleared
```

Comment:
It took me ages to understand why the comment says "cleared" but the verify is for "accept"... I think it's the problem in the code itself: we should've had an explicit clear method that doesn't accept the list of issues. 
Just a note, maybe for now it will be better to improve the comment ".. should have been cleared by passing an empty list"